include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -254,6 +256,52 @@ class two_dimensional_matrix
     }
     //!\}
 
+    /*!\brief Explicit transformation to the other major-order.
+     * \tparam other_value_t The target value type; must be assignable from `value_t`.
+     * \tparam other_allocator_t The allocator type used for the target matrix.
+     * \tparam other_order The other seqan3::detail::matrix_major_order; must not be the same as `order`.
+     *
+     * \details
+     *
+     * Copies the matrix cell by cell, rearranging the stored elements in the internal memory to represent the
+     * converted major-order.
+     *
+     * Consider the following matrix:
+     *
+     * 0  1  2  3
+     * 4  5  6  7
+     * 8  9  10 11
+     *
+     * In row-major-order the data is stored in a flat vector in the following way:
+     * 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+     *
+     * Converting it to column-major-order will rearrange the elements:
+     * 0, 4, 8, 1, 5, 9, 2, 6, 10, 3, 7, 11
+     *
+     * Note that the matrix is not transposed, such that the general layout as displayed above will remain the same.
+     * Only when streaming the data with the iterator the access order is changed.
```

Comment:
```suggestion
     * It only changes the order of iterating over the data.
```

hmm.. but theoretically, if I change the order from column to row, and still "treat" it as column order, I do get a transposed matrix, don't I?

Reply:
No, transposing means that the first column becomes the first row etc. But the major order is used for the internal storage layout since the data is flattened in a one dimensional vector. It means, that if I do ++ on the iterator over the matrix do I go over the rows or over the columns. But the the value at a given cell [i][j] does not change.